src/client/Microsoft.Identity.Client/Cache/CacheKeyFactory.cs

```diff
@@ -48,12 +50,17 @@ public static string GetExternalCacheKeyFromResponse(
                 return key;
             }
 
-            if (requestParameters.AppConfig.IsConfidentialClient ||
-                requestParameters.ApiId == ApiEvent.ApiIds.AcquireTokenSilent)
+            if (requestParameters.AppConfig.IsConfidentialClient)
```

Comment:
This is incorrect. For `AcquireTokenForClient` it will return at line 50. 

It also shows that there is insufficient testing. The `TokenNotificationsArgs` have a `SuggestedCacheKey` property that needs to checked.

Reply:
Yeah, I needed to validate the external key. Updated